### PR TITLE
Fix for inflation artifacts behind update bounds

### PR DESCRIPTION
**Pull request by AlexeyMerzlyakov**
This appearing when some layer is going after `inflation_layer`. More details are in #2051 ticket.

### CODE REVIEW

**Comment by AlexeyMerzlyakov** [comment]
ToDo: fix failing tests

**Review by SteveMacenski — COMMENTED** [review]
This makes sense to me - though its unclear to me why CI is failing (though 1 is linting)

**Comment by AlexeyMerzlyakov** [comment]
> This makes sense to me - though its unclear to me why CI is failing (though 1 is linting)

Great. Regarding failed test cases: in inflation_tests `updateCosts()` call boundaries did not cover full inflated area. Previously did worked as inflation spreaded out of bounds. Currently became to fail in test. Fixed in https://github.com/ros-planning/navigation2/pull/2259/commits/8ed97de8316e75128f2a6414e0c222cf8af4d11f.
Linting test was failed due to incorrect indent style I've missed. Also was fixed in the same commit.

File: nav2_costmap_2d/test/integration/inflation_tests.cpp [hunk 1]
```diff
@@ -328,7 +328,7 @@ TEST_F(TestNode, testInflationAroundUnkown)
   layers.updateMap(0, 0, 0);
 
   layers.getCostmap()->setCost(4, 4, nav2_costmap_2d::NO_INFORMATION);
-  ilayer->updateCosts(*layers.getCostmap(), 0, 0, 8, 8);
+  ilayer->updateCosts(*layers.getCostmap(), 0, 0, 10, 10);
```
Comment (SteveMacenski) [inline nav2_costmap_2d/test/integration/inflation_tests.cpp:331]: Why was this change required? is this working around an actual issue?

**Review by SteveMacenski — APPROVED** [review]
LGTM, anything more before its merged?

**Comment by AlexeyMerzlyakov** [comment]
Would like to fix "wings" artifacts too: they can cause noticeable troubles for keepout filter on some cases. However, there is no trivial solution I see for it yet, as we cannot make seamless stitching of window boundaries and exclude keepout zones from obstacles in saved costmaps at the same time. Still thinking about possible workaround for that. If I will find any seems to be suitable clue, I'll open a new PR for that.
